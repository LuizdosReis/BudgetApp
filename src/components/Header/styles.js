import styled from 'styled-components';

export const Header = styled.header`
  margin: 0;
  color: #333;
  text-align: center;
  padding: 2.5em 2em 0;
  border-bottom: 1px solid #eee;
`;

export const Title = styled.h1`
  margin: 0.2em 0;
  font-size: 3em;
  font-weight: 300;
`;
