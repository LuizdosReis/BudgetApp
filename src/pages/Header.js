import React from 'react';
import PropTypes from 'prop-types';

const Header = ({ title }) => (
  <header className="header">
    <h1 className="header__title">{title}</h1>
  </header>
);

Header.propTypes = {
  title: PropTypes.string.isRequired,
};

export default Header;
