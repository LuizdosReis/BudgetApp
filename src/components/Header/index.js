import React from 'react';

import { Header, Title } from './styles';

const HeaderPage = () => (
  <Header>
    <Title>Budget</Title>
  </Header>
);

export default HeaderPage;
