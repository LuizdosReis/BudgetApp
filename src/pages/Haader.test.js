import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './Header';

test('should render h1 with title', () => {
  const wrapper = shallow(<Header title="Title" />);
  expect(wrapper.find('h1').length).toBe(1);
  expect(wrapper.find('h1').text()).toBe('Title');
});

test('should render Header correctly', () => {
  const wrapper = shallow(<Header title="Title" />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
