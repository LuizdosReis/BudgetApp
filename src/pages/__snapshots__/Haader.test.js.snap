// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <h1
    className="header__title"
  >
    Title
  </h1>
</header>
`;
